src/index.js

```diff
@@ -340,6 +345,15 @@ export {
  * Only available if the underlying client isn't a ReadonlyCardanoClient
  */
 
+/**
+ * @typedef {Pick<
+ *      CardanoClient,
+ *      "isMainnet" | "submitTx" | "getTx" | "hasUtxo"
+ * >} CardanoTxSubmitter
```

Comment:
You don't have to change this: but I think the doc-gen will create something that isn't very user-friendly.

Reply:
Hm, hard to see what's better - CardanoTxSubmitter as an interface definition with duplicated props would solve one issue and raise another... Maybe with a type-assertion of matchiness on those keys between the two types, we could catch them going out of sync.

If there's a pain caused by this version, we can take some more effort ?